response_operations_ui/views/messages.py

```diff
@@ -121,14 +121,46 @@ def view_messages():
         return render_template("messages.html", breadcrumbs=breadcrumbs, response_error=True)
 
 
+@messages_bp.route('/threads/<thread_id>', methods=['GET'])
+@login_required
+def view_conversation(thread_id):
+    try:
+        thread_conversation = message_controllers.get_conversation(thread_id)['messages']
+        refined_thread = [_refine(thread) for thread in reversed(thread_conversation)]
+    except KeyError as ex:
+        logger.exception("A key error occurred")
+        raise ApiError(ex)
+    breadcrumbs = [
+        {"title": "Messages", "link": "/messages"},
+        {"title": refined_thread[0]["subject"]}
+    ]
+
+    return render_template("conversation-view.html", breadcrumbs=breadcrumbs, messages=refined_thread)
+
+
+def _get_message_subject(thread):
+    try:
+        subject = thread["subject"]
+        return subject
+    except KeyError:
+        logger.exception("Failed to retrieve Subject from thread")
+        return None
+
+
 def _refine(message):
     return {
-        'ru_ref': message.get('ru_id'),
+        'ru_ref': message.get('@ru_id').get('sampleUnitRef'),
+        'thread_id': message.get('thread_id'),
         'business_name': message.get('@ru_id').get('name'),
-        'subject': message.get('subject'),
+        'subject': _get_message_subject(message),
```

Comment:
If we don't care about logging an exception, we could replace this method with `message.get('subject')`